### ISSUE TITLE
Title of dashboard sections not rendered as <h1>

### ISSUE THREAD

**Issue by srmgh** [home-assistant/frontend#54047]
### Checklist

- [x] I have updated to the latest available Home Assistant version.
- [x] I have cleared the cache of my browser.
- [x] I have tried a different browser to see if it is related to my browser.
- [x] I have tried reproducing the issue in [safe mode](https://www.home-assistant.io/blog/2023/11/01/release-202311/#restarting-into-safe-mode) to rule out problems with unsupported custom resources.

### Describe the issue you are experiencing

Contents of a dashboard section title render inside `<p></p>` tags inside `<hui-heading-card>` tags.

### Describe the behavior you expected

w3 standards recommend using headings to convey document structure: https://www.w3.org/WAI/tips/writing/#use-headings-to-convey-meaning-and-structure

Visually impaired users rely on these headings to quickly navigate documents. The Title element of each dashboard section should probably be rendered as an `<h1>` or `<h2>` instead of inside `<p>` tags. Subtitles should render inside `<h2>` or `<h3>` tags. 

This would allow users with screen readers to quickly navigate dashboard sections and only dive in to the details once they found the section they're looking for.

The card headers for Entities cards currently do the right thing.

### Steps to reproduce the issue

1. Create a new section in a dashboard.
2. Turn on a screen reader (Talkback on Android or VoiceOver on macOS/iOS).
3. Turn off your screen
4. Navigate the dashboard using your screen reader. You'll notice it doesn't treat section headers as high level elements. This is because the screen reader is looking for `<h1>`, `<h2>`, etc. tags to use as navigational elements.

### What version of Home Assistant Core has the issue?

All of them. This is a longstanding issue.

### What was the last working version of Home Assistant Core?

It has worked correctly, as far as I'm aware.

### In which browser are you experiencing the issue?

It's not specific to browsers. Reproducible on Chrome, Safari, Brave, Firefox, etc.

### Which operating system are you using to run this browser?

macOS Tahoe, iOS. This is not an OS-specific issue. It has to do with the HTML elements used.

### State of relevant entities

```txt

```

### Problem-relevant frontend configuration

```yaml
type: heading
heading_style: title
icon: ''
badges:
  - type: entity
    show_state: true
    show_icon: true
    entity: sensor.samsung_range_tp1x_da_ks_range_0102x_machine_state
  - type: entity
    show_state: true
    show_icon: true
    entity: sensor.samsung_range_tp1x_da_ks_range_0102x_temperature
heading: Stove
```

### JavaScript errors shown in your browser console/inspector

```txt

```

### Additional information

This should be a really straightforward fix that would make blind users' experience of Home Assistant a LOT better.

**Comment by karwosts** [comment]
PR created with h2 for titles, h3 for subtitles in https://github.com/home-assistant/frontend/pull/54068
